infrastructure/azure-pipelines-dev.yml

```diff
@@ -8,38 +8,134 @@ resources:
-        displayName: Dependencies
-        env:
-          API_URI: /graphql
-          API_PROTECTED_URI: /admin/graphql
-          # We don't need Cypress in this environment, and so this avoid permission errors of installing it.
-          # See: https://docs.cypress.io/guides/getting-started/installing-cypress#Skipping-installation
-          CYPRESS_INSTALL_BINARY: 0
-
-      - task: ArchiveFiles@2
-        displayName: "Archive Files"
-        inputs:
-          rootFolderOrFile: $(System.DefaultWorkingDirectory)
-          includeRootFolder: false
-          archiveFile: "$(Build.ArtifactStagingDirectory)/Application_$(Build.BuildId).zip"
-
-      - task: PublishBuildArtifacts@1
-        displayName: "Publish Artifact: gcDigitalTalent"
-        inputs:
-          ArtifactName: gcDigitalTalent
+      name: "TBS Linux Self Hosted"
+    displayName: Test stage
+    dependsOn: Build
+    variables:
+      - name: npm_config_user_config
+        value: "$(System.DefaultWorkingDirectory)/.npmrc"
+      - name: dirResults
+        value: "$(System.DefaultWorkingDirectory)/apps/playwright/test-results"
+      - name: resourceGroupName
+        value: "rg_tbs_talent_2_PaaS"
+      - name: vertical
+        value: "dev"
+      - name: storageAccountName
+        value: "sttalclddevcacstdl"
+      - name: storageContainer
+        value: "playwright-results"
+    jobs:
+      - job: run_playwright_tests
+        displayName: Run Playwright tests
+        timeoutInMinutes: 30
+        pool:
+          name: "TBS Linux Self Hosted"
+        steps:
+          - checkout: self
+            clean: true
+            fetchDepth: 1
+
+          - task: DownloadBuildArtifacts@0
+            inputs:
+              artifactName: gcDigitalTalent
+              downloadPath: $(System.DefaultWorkingDirectory)/artifacts
+          - task: AzureCLI@2
+            displayName: "Update NPMRC Config"
+            inputs:
+              scriptType: bash
+              scriptLocation: inlineScript
+              azureSubscription: $(serviceConnectionName)
+              inlineScript: |
+                NPMRC_FILE="${{ variables.npm_config_user_config }}"
+                echo "Editing .npmrc: ${NPMRC_FILE}"
+                echo "registry=https://pkgs.dev.azure.com/tbs-sct/53c8b022-82b7-4031-88aa-0d1266fc3bf5/_packaging/TalentCloud_Custom_NPM_Feed/npm/registry/" >> "${NPMRC_FILE}"
+          - task: npmAuthenticate@0
+            displayName: "00 - NPM Authenticate"
+            name: "NPM_Authenticate"
+            inputs:
+              workingFile: $(npm_config_user_config)
+
+          - script: echo ##vso[task.setvariable variable=NPM_CONFIG_USERCONFIG]$(npm_config_user_config)
+            displayName: "Set Custom .npmrc Location"
+
+          - script: CYPRESS_INSTALL_BINARY=/opt/cypress/13.10.0/cypress.zip pnpm install cypress
+            displayName: "Install Cypress"
+
+          - script: pnpm install --no-frozen-lockfile
```

Comment:
Why does it install dependencies and rebuild the app a second time in this stage?  Isn't that the purpose of the first stage?

Reply:
Reason is that we had to install the deps on the agent that is running the tests. There was some security issue with the Talentcloud project that forced the deps downloaded from the agent instead of from the feed. 
Same steps with another test pipeline works without having to install the deps but not with our TalentCloud project

Reply:
I will try get the convo from Daniel  related to that.

Reply:
Can we drop the first stage entirely then?  Is there any point to it if we're just going to rebuild it again?

Reply:
![image (2)](https://github.com/GCTC-NTGC/gc-digital-talent/assets/89482910/5775a5f0-9dd5-4603-ab7b-d4b689b74018)

Reply:
![image (3)](https://github.com/GCTC-NTGC/gc-digital-talent/assets/89482910/31ff89b4-2a3e-445e-886e-469792ad16df)

Reply:
> Can we drop the first stage entirely then? Is there any point to it if we're just going to rebuild it again?

would check with Daniel on that.

Reply:
yeah Daniel also feels that its redundant at this point to have that stage. Will try disconnect it first if it passes then it should be completely good to go. Thanks for catching that!

Reply:
hmm doesn't work .. complaining it misses graphql dep.

Reply:
removed build stage in [53d7f44](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9652/commits/53d7f44b00cc93c53eea97403cd94d97ef68195f) !